### PR TITLE
Correct gradient accumulation

### PR DESCRIPTION
**Pull request by hyunwoongko**
1. Accumulated gradient must be updated at once. If you update every step, the effect of accumulation is lost.
2. We need to divide the loss collected in the list into a gradient accumulation steps. Currently, you are dividing by world size.

### CODE REVIEW

File: megatron/training.py [hunk 1]
```diff
@@ -457,16 +456,19 @@ def train_step(neox_args, timers, data_iterator, model, optimizer, lr_scheduler)
...
-                raise ValueError("Must be using deepspeed to run neox")
-            timers("optimizer").stop()
-        reduced_loss = {
-            "lm_loss": reduce_losses(losses).mean()
-        }  # reduces losses across machines for logging
+                
+        # Update parameters.
+        timers("optimizer").start()
+        if neox_args.deepspeed:
+            model.step()
```
Comment (hyunwoongko) [inline megatron/training.py:463]: - https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/runtime/engine.py#L1463
- https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/runtime/engine.py#L1414

Comment (hyunwoongko) [reply megatron/training.py:463]: https://medium.com/huggingface/training-larger-batches-practical-tips-on-1-gpu-multi-gpu-distributed-setups-ec88c3e51255

**Comment by hyunwoongko** [comment]
I will reupload after more checking.
